src/common/cups_print.h

```diff
@@ -59,15 +59,12 @@ typedef struct dt_print_info_t
   dt_paper_info_t paper;
 } dt_print_info_t;
 
-// is some printers are available
-gboolean is_printer_available(void);
+// Asynchronous printer discovery, cb will be called for each printer found
```

Comment:
Looking at the cups API what we probably need is cupsEnumDests: 

https://www.cups.org/documentation.php/doc-2.0/api-cups.html#cupsEnumDests

That works with a callback so should allow us to add printers as they become available. I also wonder if we need to launch that job every time we open the print module in case printers have shown up or disappeared. Long gone are the times when the printer config was static.